docs/user/datastores/index_config.rst

```diff
@@ -604,19 +604,48 @@ the trait ``org.locationtech.geomesa.index.planning.QueryInterceptor``:
         * @param query query
         */
       def rewrite(query: Query): Unit
+
+      /**
+       * Hook to allow interception of a query after extracting the query values
+       *
+       * @param strategy query strategy
+       * @return an exception if the query should be stopped
+       */
+      def guard(strategy: QueryStrategy): Option[IllegalArgumentException] = None
     }
 
 Interceptors must have a default, no-arg constructor. The interceptor lifecycle consists of:
 
 1. The instance is instantiated via reflection, using its default constructor
 #. The instance is initialized via the ``init`` method, passing in the data store containing the simple feature type
-#. ``rewrite`` is called repeatedly
+#. ``rewrite`` and ``guard`` are called repeatedly, once for each query
```

Comment:
I think it'd be clearer as ....

 ``rewrite`` and ``guard`` are called once for each query

Maybe that's not capturing the idea that the methods are called multiple times after the implementation is instantiated though?

Reply:
changed to your wording, and noted that close is called when the datastore is disposed